oak_runtime/src/proxy.rs

```diff
@@ -283,28 +340,74 @@ impl RuntimeProxy {
             "{:?}: channel_write({}, {:?})",
             self.node_name, write_handle, msg
         );
-        let result = self.runtime.channel_write(self.node_id, write_handle, msg);
+        let result =
+            self.runtime
+                .channel_write(self.node_id, write_handle, msg, OperationPrivilege::None);
         debug!(
             "{:?}: channel_write({}, ...) -> {:?}",
             self.node_name, write_handle, result
         );
         result
     }
 
-    /// See [`Runtime::channel_read`].
+    /// Calls [`Runtime::channel_write`] using the Node's privilege.
+    pub fn channel_write_with_privilege(
+        &self,
+        write_handle: oak_abi::Handle,
+        msg: NodeMessage,
+    ) -> Result<(), OakStatus> {
+        debug!(
+            "{:?}: channel_write_with_privilege({}, {:?})",
+            self.node_id, write_handle, msg
+        );
+        let result = self.runtime.channel_write(
+            self.node_id,
+            write_handle,
+            msg,
+            OperationPrivilege::UsePrivilege,
+        );
+        debug!(
+            "{:?}: channel_write_with_privilege({}, ...) -> {:?}",
+            self.node_id, write_handle, result
+        );
+        result
+    }
+
+    /// Calls [`Runtime::channel_read`] without using the Node's privilege.
     pub fn channel_read(
         &self,
         read_handle: oak_abi::Handle,
     ) -> Result<Option<NodeMessage>, OakStatus> {
         debug!("{:?}: channel_read({})", self.node_name, read_handle,);
-        let result = self.runtime.channel_read(self.node_id, read_handle);
+        let result = self
+            .runtime
+            .channel_read(self.node_id, read_handle, OperationPrivilege::None);
         debug!(
             "{:?}: channel_read({}) -> {:?}",
             self.node_name, read_handle, result
         );
         result
     }
 
+    /// Calls [`Runtime::channel_read`] using the Node's privilege.
+    pub fn channel_read_with_privilege(
```

Comment:
Wasn't there some talk of only allowing declassifying-writes, not declassifying-reads? @tiziano88 do you recall whether there was a conclusion?

Reply:
There was a discussion in #1301 and the outcome was that it was not required for the security properties we want and might be needed in future. It is not available in the ABI and SDK at the moment, with the idea that we can add it later if we need it. I just added this to the proxy to support two existing tests in the runtime.